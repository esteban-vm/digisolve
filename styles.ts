import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
