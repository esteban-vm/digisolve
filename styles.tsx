import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
