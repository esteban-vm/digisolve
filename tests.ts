import type { ReactElement } from 'react'
import { create, type ReactTestRendererJSON, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON() as ReactTestRendererJSON
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
