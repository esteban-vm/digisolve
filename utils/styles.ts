import { css } from '@emotion/react'

export enum breakPoints {
  /** Big tablets to smaller laptop screens */
  lg = '1200px',
  /** Smaller tablets to larger tablet screens */
  md = '1023px',
  /** Larger phones to smaller tablet screens */
  sm = '767px',
  /** Smaller to larger mobile phones */
  xs = '479px',
}

export const mediaQuery = (breakPoint: keyof typeof breakPoints) => {
  return `@media only screen and (max-width: ${breakPoints[breakPoint]})`
}

export const globalStyles = css`
  @import url('/css/fluid.min.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;
    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;
    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;
    --color-dark: #888;
    --color-darker: #333;
    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;
    --color-shadow: #efefefef;
    --color-facebook: #3b5998;
    --color-twitter: #00aced;
    --color-google-plus: #dd4b39;
    --color-youtube: #f00;
    --color-instagram: #517fa4;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;

    ${mediaQuery('lg')} {
      font-size: 14px;
    }

    ${mediaQuery('md')} {
      font-size: 12px;
    }

    ${mediaQuery('sm')} {
      font-size: 10px;
    }
  }

  * {
    :focus {
      outline: none;
    }

    ::selection {
      background-color: var(--color-primary);
      color: var(--color-light);
    }

    ::-webkit-scrollbar {
      width: 0.5rem;

      &-track {
        box-shadow: inset 0 0 6px var(--color-darker);
      }

      &-thumb {
        background-color: var(--color-primary);
        outline: 1px solid var(--color-dark-primary);
        border-radius: 10px;
      }
    }
  }

  img,
  button,
  a {
    user-select: none;
  }

  ul {
    list-style: none;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;

    ${mediaQuery('md')} {
      padding: 0 2%;
    }
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }

  .section {
    &__heading {
      font-size: 250%;
      text-transform: capitalize;
      color: var(--color-black);
      margin-bottom: 20px;
      margin-top: 150px;

      ${mediaQuery('lg')} {
        margin-top: 90px;
      }

      ${mediaQuery('sm')} {
        margin-top: 60px;
        margin-bottom: 10px;
      }
    }

    &__subheading {
      font-size: 150%;
      color: var(--color-grey);
      font-weight: 300;
      margin-bottom: 30px;
      line-height: 1.5;

      ::after {
        content: ' ';
        display: block;
        height: 2px;
        width: 100px;
        background-color: var(--color-primary);
        margin: 30px auto 0;

        ${mediaQuery('sm')} {
          margin-top: 15px;
        }
      }
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
