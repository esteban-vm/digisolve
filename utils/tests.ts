import type { ReactElement } from 'react'
import { create, type TestRendererOptions } from 'react-test-renderer'

const customCreate = (ui: ReactElement, options?: TestRendererOptions) => {
  return create(ui, options).toJSON()
}

export { customCreate as create }
export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
