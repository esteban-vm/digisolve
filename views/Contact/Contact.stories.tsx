import type { Meta } from '@/stories'
import Contact from './Contact'

export default {
  component: Contact,
  title: 'Views/Contact',
  parameters: {
    backgrounds: {
      disable: true,
    },
  },
} as Meta<typeof Contact>

export const View = () => <Contact />
