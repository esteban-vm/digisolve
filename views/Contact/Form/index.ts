export { default, type FormFields } from './Form'
