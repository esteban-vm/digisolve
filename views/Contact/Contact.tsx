import type { Component } from '@/types'
import { useId } from 'react'
import { Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import Form from './Form'

const ContactComponent: Component = (props) => {
  const nameID = useId()

  return (
    <section id='contact' aria-labelledby={nameID} {...props}>
      <Grid.Row>
        <h2 id={nameID}>We are happy to hear from you</h2>
      </Grid.Row>
      <Grid.Row>
        <Form />
      </Grid.Row>
    </section>
  )
}

const Contact = styled(ContactComponent)`
  text-align: center;

  h2 {
    color: var(--color-black);
    margin-top: 150px;
    margin-bottom: 50px;

    ${mediaQuery('lg')} {
      margin-top: 90px;
    }

    ${mediaQuery('sm')} {
      margin-top: 60px;
    }
  }
`

export default Contact
