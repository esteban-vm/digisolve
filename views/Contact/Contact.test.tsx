import { render, cleanup, screen, create } from '@/tests'
import { breakPoints } from '@/styles'
import Contact from './Contact'

describe('ðŸ§ª CONTACT:', () => {
  describe('display tests:', () => {
    /** Section title */
    const name = /^we are happy to hear from you$/i

    beforeEach(() => {
      render(<Contact />)
    })

    afterEach(cleanup)

    it('should be accessible', () => {
      const section = screen.getByRole('region', { name })
      expect(section).toBeInTheDocument()
    })

    describe('should be displayed:', () => {
      it('the heading', () => {
        const heading = screen.getByRole('heading', { name, level: 2 })
        expect(heading).toBeInTheDocument()
        expect(heading).toBeVisible()
      })

      it('the form', () => {
        const form = screen.getByRole('form')
        expect(form).toBeInTheDocument()
        expect(form).toBeVisible()
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Contact />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    describe('should have correct styles on:', () => {
      it('small screen devices', () => {
        expect(tree).toHaveStyleRule('margin-top', '60px', {
          target: 'h2',
          media: `(max-width: ${breakPoints.sm})`,
        })
      })

      it('large screen devices', () => {
        expect(tree).toHaveStyleRule('margin-top', '90px', {
          target: 'h2',
          media: `(max-width: ${breakPoints.lg})`,
        })
      })
    })
  })
})
