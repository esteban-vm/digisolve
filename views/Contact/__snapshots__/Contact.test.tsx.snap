// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª CONTACT: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
}

.emotion-0 h2 {
  color: var(--color-black);
  margin-top: 150px;
  margin-bottom: 50px;
}

@media only screen and (max-width: 1200px) {
  .emotion-0 h2 {
    margin-top: 90px;
  }
}

@media only screen and (max-width: 768px) {
  .emotion-0 h2 {
    margin-top: 60px;
  }
}

.emotion-1 {
  text-align: left;
  width: 60%;
  margin: 0 auto;
  color: var(--color-grey);
}

@media only screen and (max-width: 768px) {
  .emotion-1 {
    width: 70%;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    width: 90%;
  }
}

.emotion-1 label {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

@media only screen and (max-width: 480px) {
  .emotion-1 label {
    margin-top: 5px;
  }

  .emotion-1 label::after {
    content: ':';
  }
}

.emotion-1 .field {
  width: 95%;
  padding: 8px;
  border-radius: 3px;
  border: 1px solid #ccc;
  background-color: var(--color-white);
}

.emotion-1 .field::-webkit-input-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-1 .field::-moz-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-1 .field:-ms-input-placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-1 .field::placeholder {
  text-transform: capitalize;
  opacity: 0.7;
}

.emotion-1 .field:is(textarea) {
  height: 100px;
  resize: none;
}

.emotion-1 .check {
  margin: 10px 5px 10px 0;
}

@media only screen and (max-width: 480px) {
  .emotion-1 .check {
    margin-top: 2px;
  }
}

.emotion-1 .toast {
  text-align: center;
  color: var(--color-primary);
  background-color: var(--color-light);
}

.emotion-1 .icon {
  color: var(--color-secondary);
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 input[type='submit'],
.emotion-2 a {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  padding: 10px 30px;
  border-radius: 200px;
  font-weight: 500;
  outline: none;
  -webkit-transition: background-color 500ms;
  transition: background-color 500ms;
  border: 2px solid var(--color-primary);
  background-color: var(--color-primary);
  color: var(--color-white);
}

.emotion-2 input[type='submit']:hover,
.emotion-2 a:hover,
.emotion-2 input[type='submit']:active,
.emotion-2 a:active {
  background-color: var(--color-dark-primary);
}

<section
  aria-labelledby="contact_title"
  className="emotion-0"
  id="contact"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      id="contact_title"
    >
      We are happy to hear from you
    </h2>
  </div>
  <div
    className="row "
    role="presentation"
  >
    <form
      aria-label="Send a message"
      className="emotion-1"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="row "
        role="presentation"
      >
        <div
          className="col span_1_of_3 "
          role="presentation"
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
        </div>
        <div
          className="col span_2_of_3 "
          role="presentation"
        >
          <input
            autoCorrect="off"
            className="field"
            id="name"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="your name"
            spellCheck={false}
            type="text"
          />
        </div>
      </div>
      <div
        className="row "
        role="presentation"
      >
        <div
          className="col span_1_of_3 "
          role="presentation"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
        </div>
        <div
          className="col span_2_of_3 "
          role="presentation"
        >
          <input
            className="field"
            id="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="your email"
            type="email"
          />
        </div>
      </div>
      <div
        className="row "
        role="presentation"
      >
        <div
          className="col span_1_of_3 "
          role="presentation"
        >
          <label
            htmlFor="newsletter"
          >
            Newsletter?
          </label>
        </div>
        <div
          className="col span_2_of_3 "
          role="presentation"
        >
          <input
            className="check"
            defaultChecked={true}
            id="newsletter"
            name="newsletter"
            onBlur={[Function]}
            onChange={[Function]}
            type="checkbox"
          />
          Yes, Please
        </div>
      </div>
      <div
        className="row "
        role="presentation"
      >
        <div
          className="col span_1_of_3 "
          role="presentation"
        >
          <label
            htmlFor="message"
          >
            Drop us a line
          </label>
        </div>
        <div
          className="col span_2_of_3 "
          role="presentation"
        >
          <textarea
            className="field"
            id="message"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="your message"
          />
        </div>
      </div>
      <div
        className="row "
        role="presentation"
      >
        <div
          className="col span_1_of_3 "
          role="presentation"
        />
        <div
          className="col span_2_of_3 "
          role="presentation"
        >
          <span
            className="emotion-2"
          >
            <input
              type="submit"
              value="Send it!"
            />
          </span>
        </div>
      </div>
      <div
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          {
            "bottom": 16,
            "left": 16,
            "pointerEvents": "none",
            "position": "fixed",
            "right": 16,
            "top": 16,
            "zIndex": 9999,
          }
        }
      />
    </form>
  </div>
</section>
`;
